src/main/groovy/nebula/plugin/publishing/maven/ResolvedMavenPlugin.groovy

```diff
@@ -51,6 +51,10 @@ class ResolvedMavenPlugin implements Plugin<Project> {
                         [scope, resolutionMap]
                     }
 
+                    // Another plugin may have already patched the POM to support 'provided' scope.
+                    // For the purpose of version resolution, 'provided' scope is equavalent to 'runtime'.
+                    scopeResolutionMap['provided'] = scopeResolutionMap['runtime']
```

Comment:
The order of plugin application will not matter, unless the other plugin does something exotic like adding a dependency to the `runtime` configuration after the project has been evaluated (in which case the `maven-publish` plugin doesn't even include the dependency in the POM, so it shouldn't really be an issue).

With this change, `ResolvedMavenPlugin` assumes any dependencies marked as `<scope>provided</scode>` in the POM are members of the `runtime` configuration, which is likely to be true in common usage. If a certain plugin implements `provided` differently, the worst that happens is that the nebula plugin will not resolve dynamic versions for the provided dependencies.